utils/nctl/sh/scenarios/itst02.sh

```diff
@@ -0,0 +1,165 @@
+#!/usr/bin/env bash
+
+source "$NCTL"/sh/utils/main.sh
+source "$NCTL"/sh/views/utils.sh
+source "$NCTL"/sh/node/svc_"$NCTL_DAEMON_TYPE".sh
+source "$NCTL"/sh/utils/infra.sh
+
+# Exit if any of the commands fail.
+set -e
+
+#######################################
+# Runs an integration tests that tries to simulate
+# stalling consensus by stopping enough nodes. It
+# then restarts the nodes and checks for the chain
+# to progress.
+#
+# Arguments:
+#   `timeout=XXX` timeout (in seconds) when syncing.
+#######################################
+function main() {
+    log "------------------------------------------------------------"
+    log "Starting Scenario: itst02"
+    log "------------------------------------------------------------"
+
+    # 0. Wait for network start up
+    do_await_genesis_era_to_complete
+    # 1. Allow chain to progress
+    do_await_era_change
+    # 2-4. Stop three nodes
+    do_stop_node "5"
+    do_stop_node "4"
+    do_stop_node "3"
+    # 5. Ensure chain stalled
+    sleep_and_compare "60"
+    # 6-8. Restart three nodes
+    do_start_node "5"
+    do_start_node "4"
+    do_start_node "3"
+    # 9-11. Check sync of restarted node
+    do_await_full_synchronization "5"
+    do_await_full_synchronization "4"
+    do_await_full_synchronization "3"
+    # 12. Ensure era proceeds after restart
+    do_await_era_change "2"
+    # 13-15. Check sync of nodes again
+    do_await_full_synchronization "5"
+    do_await_full_synchronization "4"
+    do_await_full_synchronization "3"
+    # 16-18. Compare stalled lfb hash to current
+    compare "5" "$STALLED_LFB"
+    compare "4" "$STALLED_LFB"
+    compare "3" "$STALLED_LFB"
+
+    log "------------------------------------------------------------"
+    log "Scenario itst02 complete"
+    log "------------------------------------------------------------"
+}
+
+function log_step() {
+    local COMMENT=${1}
+    log "------------------------------------------------------------"
+    log "STEP $STEP: $COMMENT"
+    STEP=$((STEP + 1))
+}
+
+function compare() {
+    log_step "node-${1}: checking chain progressed"
+    local LFB1=$(do_read_lfb_hash ${1})
+    local LFB2=${2}
+
+    if [ "$LFB1" = "$LFB2" ]; then
+       log "error: $LFB1 = $LFB2, chain didn't progress."
+       exit 1
+    fi
+}
+
+function sleep_and_compare() {
```

Comment:
This function is very similar to `compare` but the assertion is inverted. Can we have a separate function `assert_chain_stalled` and call it from the `sleep` here?

Reply:
you are better at names 😄 I'll update it